lib/fog_extensions/libvirt/server.rb

```diff
@@ -50,8 +50,8 @@ def interfaces
 
       def select_nic(fog_nics, nic)
         nic_attrs = nic.compute_attributes
-        match =   fog_nics.detect { |fn| fn.network == nic_attrs['network'] } # grab any nic on the same network
-        match ||= fog_nics.detect { |fn| fn.bridge  == nic_attrs['bridge']  } # no network? try a bridge...
+        match =   fog_nics.detect { |fn| !fn.network.nil? && fn.network == nic_attrs['network'] } # grab any nic on the same network
+        match ||= fog_nics.detect { |fn| !fn.bridge.nil? && fn.bridge  == nic_attrs['bridge'] } # no network? try a bridge...
```

Comment:
How about avoiding the whole loop?
```suggestion
        match = nil
        match = fog_nics.detect { |fn| fn.network == nic_attrs['network'] } unless nic_attrs['network'].nil? # grab any nic on the same network
        match ||= fog_nics.detect { |fn| fn.bridge == nic_attrs['bridge'] } unless nic_attrs['bridge'].nil? # no network? try a bridge...
```

Reply:
Will preface this with we don't do much with ruby so there are usually better ways to do what we want/are trying to write.

From some quick testing, the `unless` on the second line seems to block the `or-equals` entirely which I wasn't expecting.  I do agree skipping the loop entirely makes sense.  The best I have is wrapping each line in an `if` which seems to get the correct behavior here and preserves the `or-equals`.  Pending your thoughts will push an updated version.